homeassistant/components/light/mochad.py

```diff
@@ -47,17 +48,19 @@ class MochadLight(Light):
 
     def __init__(self, hass, ctrl, dev):
         """Initialize a Mochad Light Device."""
-        from pymochad import device
-
         self._controller = ctrl
         self._address = dev[CONF_ADDRESS]
         self._name = dev.get(CONF_NAME,
                              'x10_light_dev_{}'.format(self._address))
         self._comm_type = dev.get(mochad.CONF_COMM_TYPE, 'pl')
-        self.light = device.Device(ctrl, self._address,
-                                   comm_type=self._comm_type)
+
         self._brightness = 0
-        self._state = self._get_device_status()
+        # Init with false to avoid locking HA for long on CM19A (goes from rf
+        # to pl via TM751, but not other way around)
+        if self._comm_type == 'pl':
+            self._state = self._get_device_status()
```

Comment:
We don't want side effects in init method. Define an `update` method and move this call to that method. Then call `add_entities` like so:
```py
add_entities(entities, True)
```
Also set `should_poll` to `False`.

Reply:
I tried adding it. After testing - removed again. It was causing switches to go back to the previous state in the UI.

Reply:
We have to take control of the state update, when setting `should_poll` to `False`. If there's no quick feedback possible, we have to schedule a state update in the actuator methods.
Did you try adding the `update` method? Setting `should_poll` to `False` was only the consequence of the former.

Reply:
We can't add a feature and break rules in one PR , and then fix that in a second PR. We should follow the rules from the beginning.

Reply:
It would be ok to make a new PR that refactor the component first, so that it's easier to add this feature in a second PR without breaking rules, after the first PR.